src/sage/combinat/k_regular_sequence.py

```diff
@@ -101,6 +101,114 @@
 from sage.misc.cachefunc import cached_function, cached_method
 
 
+def pad_right(T, length, zero=0):
+    r"""
+    Pad ``T`` to the right by using ``zero`` to have
+    at least the given ``length``.
+
+    INPUT:
+
+    - ``T`` -- A tuple, list or other iterable
+
+    - ``length`` -- a nonnegative integer
+
+    - ``zero`` -- (default: ``0``) the elements to pad with
+
+    OUTPUT:
+
+    An object of the same type as ``T``
+
+    EXAMPLES::
+
+        sage: from sage.combinat.k_regular_sequence import pad_right
+        sage: pad_right((1, 2, 3), 10)
+        (1, 2, 3, 0, 0, 0, 0, 0, 0, 0)
+        sage: pad_right((1, 2, 3), 2)
+        (1, 2, 3)
+
+    TESTS::
+
+        sage: pad_right([1, 2, 3], 10)
+        [1, 2, 3, 0, 0, 0, 0, 0, 0, 0]
+    """
+    return T + type(T)(zero for _ in range(length - len(T)))
+
+
+def value(D, k):
+    r"""
+    Return the value of the expansion with digits `D` in base `k`, i.e.
+
+    .. MATH::
+
+        \sum_{0\leq j < \operator{len}D} D[j] k^j.
+
+    INPUT:
+
+    - ``D`` -- a tuple or other iterable
+
+    - ``k`` -- the base
+
+    OUTPUT:
+
+    An element in the common parent of the base `k` and of the entries
+    of `D`
+
+    EXAMPLES::
+
+        sage: from sage.combinat.k_regular_sequence import value
+        sage: value(42.digits(7), 7)
+        42
+    """
+    return sum(d * k**j for j, d in enumerate(D))
+
+
+def split_interlace(n, k, p):
+    r"""
+    Split each digit in the `k`-ary expansion of `n` into `p` parts and
+    return the value of the expansion obtained by each of these parts.
```

Comment:
From the description, I do not understand what "split a digit into p parts" shall mean. Instead of the many examples (which might be moved to TESTS) I'd prefer to see fewer examples with an explanation what the output means. Perhaps "split into p parts" could be made explicit into saying that it will be expanded as a k/p-ary expansion? And also say that "value of the expansion" means expansion in base k/p?

https://github.com/sagemath/sage/issues/21204#issuecomment-1417820841 Item 2